package/MDAnalysis/core/selection.py

```diff
@@ -1381,3 +1356,55 @@ def _parse_subexp(self):
 
 # The module level instance
 Parser = SelectionParser()
+
+# create a module container to avoid name clashes of autogenerated classes
+selection_classes = types.ModuleType(f'{__name__}.selection_classes')
+# stick it in sys.modules so pickle can find it
+sys.modules[selection_classes.__name__] = selection_classes
+
+
+def gen_selection_class(singular, attrname, dtype, per_object):
+    """Class factory for arbitrary TopologyAttrs
+
+    Parameters
+    ----------
+    singular: str
+        singular name of TopologyAttr
+    attrname: str
+        attribute name of TopologyAttr
+    dtype:
+        type of TopologyAttr
+    per_object:
+        level of TopologyAttr
+
+    Returns
+    -------
+    selection: subclass of Selection
```

Comment:
Can you add some explanation to explain the general idea of automatically generating selections?

I guess this function is a good place as it is the key component that needs to be called. However, there should also be some documentation in `topologyattrs` that links here. 

My concern is that, even though your code is commented, it will soon become obscure what magic is happening and why. I am not pushing for filling the empty https://docs.mdanalysis.org/2.0.0-dev0/documentation_pages/topology.html page but rather leaving some coherent writing somewhere in the docs so that we have a fighting chance to pull it together one day.

Reply:
I added docstrings to `gen_selection_class` and `_TopologyAttrMeta`. The latter won't be visible to users as private things aren't autodocumented. Otherwise I'm not sure how best to treat this other than adding

* a script-generated table of available tokens in the User Guide (possibly also the normal docs)
* a short section on how selection classes + tokens work in the advanced section of the User Guide / normal docs

<img width="571" alt="Screen Shot 2020-09-23 at 2 47 44 pm" src="https://user-images.githubusercontent.com/31115101/93968007-d7348600-fdab-11ea-839a-f7a8dc61f92e.png">